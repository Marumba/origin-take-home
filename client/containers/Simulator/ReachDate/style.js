import styled from '@emotion/styled';

export const ReachDate = styled.div`
	${({ customCss, theme }) => (typeof customCss === 'function' ? customCss(theme) : customCss)}
`;
