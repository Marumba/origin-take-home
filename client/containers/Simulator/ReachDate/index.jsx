/* eslint-disable */
import React, { useCallback } from 'react';
import { atom, useRecoilState } from 'recoil';
import { string, customCssType } from '~/types';

import { Input } from '~/components/forms';

import { MONTHS } from '~/helpers/constants';
import { getYearMonthDatetime } from '~/helpers/utils';

import * as S from './style';

const REACH_DATE_ATOM_ID = 'simulator-reach-date';

export const reachDateState = atom({
	key: REACH_DATE_ATOM_ID,
	default: {},
	effects_UNSTABLE: [
		({ onSet, setSelf }) => {
			const storedReachDate = localStorage.getItem(REACH_DATE_ATOM_ID);
			if (storedReachDate !== null) {
				const localAmount = JSON.parse(storedReachDate);
				const { datetime } = getYearMonthDatetime();
				if (datetime < new Date(localAmount.date)) setSelf(JSON.parse(storedReachDate));
			}
			onSet(newReachDate => {
				localStorage.setItem(REACH_DATE_ATOM_ID, JSON.stringify(newReachDate));
			});
		}
	]
});
function ReachDate({ customCss, ...inputProps }) {
	const [reachDate, setReachDate] = useRecoilState(reachDateState);

	const handleChange = useCallback(
		(date, months, state) => {
			const year = date.slice(0, 4);
			const month = +date.slice(5) - 1;
			const namedMonth = MONTHS[month];
			setReachDate({ date: `${namedMonth} ${year}`, months, state });
		},
		[setReachDate]
	);

	return (
		<S.ReachDate data-testid="simulator-reach-date" customCss={customCss}>
			<Input.MonthPicker hidden onChange={handleChange} localValue={reachDate.state} {...inputProps} />
		</S.ReachDate>
	);
}

ReachDate.propTypes = {
	customCss: customCssType,
	label: string
};

ReachDate.defaultProps = {
	customCss: undefined,
	label: ''
};

export default ReachDate;
