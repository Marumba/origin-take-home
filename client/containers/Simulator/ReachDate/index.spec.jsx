import React from 'react';

import { render, screen } from '~/helpers/testing.customRtl';

import ReachDate from '.';

describe('Container Simulator ReachDate', () => {
	it('should render', () => {
		const { container } = render(<ReachDate id="nededed" name="nededed" />);
		expect(screen.getByTestId('simulator-reach-date')).toBeInTheDocument();

		expect(container.firstChild).toMatchSnapshot();
	});

	it('should incorporate de given css', () => {
		render(
			<ReachDate
				id="nededed"
				name="nededed"
				customCss={{
					background: 'red'
				}}
			/>
		);
		expect(screen.getByTestId('simulator-reach-date')).toHaveStyleRule('background', 'red');
	});
});
