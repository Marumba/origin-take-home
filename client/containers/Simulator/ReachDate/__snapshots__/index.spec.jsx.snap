// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Simulator ReachDate should render 1`] = `
.emotion-2:focus {
  background: red;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-4 input {
  width: auto;
  -webkit-transform: translateX(-50%);
  -moz-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  position: absolute;
  left: 50%;
  z-index: 1;
}

.emotion-6 {
  font-family: 'Work Sans',sans-serif;
  font-weight: 400;
  font-size: 12px;
  line-height: 150%;
  margin-bottom: 4px;
  color: #1E2A32;
}

@media screen and (min-width: 992px) {
  .emotion-6 {
    font-size: 14px;
  }
}

.emotion-8 {
  border: 1px solid #E9EEF2;
  border-radius: 4px;
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 56px;
  position: relative;
  padding: 0 14px;
}

.emotion-8 svg+input {
  padding-left: 7px;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  width: 50%;
  height: 56px;
  padding: 0 14px;
  position: absolute;
  z-index: 10;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  left: 0;
}

@media screen and (min-width: 992px) {
  .emotion-10 {
    padding: 0;
  }

  .emotion-10 svg {
    margin-left: 12px;
  }

  .emotion-10:hover svg {
    -webkit-transform: scale(1.2);
    -moz-transform: scale(1.2);
    -ms-transform: scale(1.2);
    transform: scale(1.2);
  }
}

.emotion-10:active svg {
  -webkit-transform: scale(1.4);
  -moz-transform: scale(1.4);
  -ms-transform: scale(1.4);
  transform: scale(1.4);
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

.emotion-12 {
  border: 1px solid #E9EEF2;
  border-radius: 4px;
  background: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 56px;
  border: 0;
  cursor: pointer;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 5;
}

.emotion-14 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Work Sans',sans-serif;
  font-weight: 400;
  font-size: 14px;
  line-height: 150%;
}

@media screen and (min-width: 992px) {
  .emotion-14 {
    font-size: 16px;
  }
}

.emotion-14.semibold,
.emotion-14 strong {
  font-weight: 600;
}

.emotion-14.highEmphasis {
  color: #1E2A32;
}

.emotion-14.lowEmphasis {
  color: #708797;
}

.emotion-16 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  font-family: 'Work Sans',sans-serif;
  font-weight: 400;
  font-size: 14px;
  line-height: 150%;
}

@media screen and (min-width: 992px) {
  .emotion-16 {
    font-size: 16px;
  }
}

.emotion-16.semibold,
.emotion-16 strong {
  font-weight: 600;
}

.emotion-16.highEmphasis {
  color: #1E2A32;
}

.emotion-16.lowEmphasis {
  color: #708797;
}

.emotion-18 {
  font-family: 'Rubik',sans-serif;
  font-weight: 500;
  font-size: 20px;
  line-height: 120%;
  color: #4D6475;
  padding: 14px 0;
  width: 100%;
}

@media screen and (min-width: 992px) {
  .emotion-18 {
    font-size: 24px;
  }
}

.emotion-18.highEmphasis {
  color: #1E2A32;
}

.emotion-20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  width: 50%;
  height: 56px;
  padding: 0 14px;
  position: absolute;
  z-index: 10;
  -webkit-transition: -webkit-transform 100ms ease-in-out;
  transition: transform 100ms ease-in-out;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
  right: 0;
}

@media screen and (min-width: 992px) {
  .emotion-20 {
    padding: 0;
  }

  .emotion-20 svg {
    margin-left: 12px;
  }

  .emotion-20:hover svg {
    -webkit-transform: scale(1.2);
    -moz-transform: scale(1.2);
    -ms-transform: scale(1.2);
    transform: scale(1.2);
  }
}

.emotion-20:active svg {
  -webkit-transform: scale(1.4);
  -moz-transform: scale(1.4);
  -ms-transform: scale(1.4);
  transform: scale(1.4);
  -webkit-transition: 0.1s;
  transition: 0.1s;
}

@media screen and (min-width: 992px) {
  .emotion-20 svg {
    margin-left: 0;
    margin-right: 12px;
  }

  .emotion-20:hover svg {
    -webkit-transform: rotate(180deg) scale(1.2);
    -moz-transform: rotate(180deg) scale(1.2);
    -ms-transform: rotate(180deg) scale(1.2);
    transform: rotate(180deg) scale(1.2);
  }
}

.emotion-20:active svg {
  -webkit-transform: rotate(180deg) scale(1.4);
  -moz-transform: rotate(180deg) scale(1.4);
  -ms-transform: rotate(180deg) scale(1.4);
  transform: rotate(180deg) scale(1.4);
}

<div
  class="emotion-0 emotion-1"
  data-testid="simulator-reach-date"
>
  <div
    class="emotion-2 emotion-3"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <label
        class="emotion-6 emotion-7"
        for="nededed"
      />
      <div
        class="emotion-8 emotion-9"
      >
        <button
          class="emotion-10 emotion-11"
          type="button"
        >
          <div />
        </button>
        <div
          class="emotion-12 emotion-13"
        >
          <div
            class="highEmphasis semibold emotion-14 emotion-15"
          >
            September
          </div>
          <div
            class="highEmphasis emotion-16 emotion-17"
          >
            2021
          </div>
        </div>
        <input
          class="emotion-18 emotion-19"
          hidden=""
          id="nededed"
          name="nededed"
          readonly=""
          type="text"
          value="2021-9"
        />
        <button
          class="emotion-20 emotion-21"
          type="button"
        >
          <div
            transform="rotate(180)"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
